pinecone/index_connection.go

```diff
@@ -0,0 +1,364 @@
+package pinecone
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/pinecone-io/go-pinecone/internal/gen/data"
```

Comment:
Yeah, the import/package system in Go works like this. They decided on a "distributed" package manager instead of centralized. The `go.mod` file at root defines this module as `github.com/pinecone-io/go-pinecone`. When it sees that prefix, it's local to this module. Also, `internal` can only be imported by files in the same package.